### PR TITLE
updated NullIndicator to convert to int8, fixed tests

### PR DESCRIPTION
**Pull request by limlam96**
Updating NullIndicator to coerce return to int8. Return cols will be all 0/1, so can't see a downside to this, and will give large memory savings to outputs.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -16,6 +16,13 @@ Subsections for each version can be one of the following;
 
 Each individual change should have a link to the pull request after the description of the change.
 
+1.1.2 (2024-02-07)
+------------------
+
+Added
+^^^^^
+- Updated NullIndicator to return int8 columns https://github.com/lvgig/tubular/pull/173
```
Comment (davidhopkinson26) [inline CHANGELOG.rst:24]: please update to the format in rest of changelog (look at the raw markdown, I cant copy it in here without rendering messing with the code!)

**Review by davidhopkinson26 — COMMENTED** [review]
Looks good.  just one comment on the changelog update

File: CHANGELOG.rst (hunk 1 above)
Comment (limlam96) [reply CHANGELOG.rst]: hopefully that's fixed here? https://github.com/lvgig/tubular/pull/173/commits/0651eafb7f6317751751354e9795cb7f2be06c57

**Review by davidhopkinson26 — APPROVED** [review]
All looks good

**Pull request merged**